TestProjects/ShaderFoundry/Assets/CommonAssets/Editor/FoundryTestRenderer.cs

```diff
@@ -37,8 +37,21 @@ internal void ReportTests()
     internal Vector3 testPosition = new Vector3(0.24699998f, 0.51900005f, 0.328999996f);
     internal Quaternion testRotation = new Quaternion(-0.164710045f, -0.0826543793f, -0.220811233f, 0.957748055f);
 
+    internal void CheckForShaderErrors(Shader shader)
+    {
+        if (ShaderUtil.ShaderHasError(shader))
+        {
+            var messages = ShaderUtil.GetShaderMessages(shader);
+            foreach (var message in messages)
+                if (message.severity == UnityEditor.Rendering.ShaderCompilerMessageSeverity.Error)
```

Comment:
Do we want to check for warnings as well?

Reply:
I'll add a todo. This was something I added to help diagnose what went wrong when making the tests (did the test fail or did the shader just not compile?). We may have warnings right now just due to the SRP template stuff.